### ISSUE TITLE
Use Node.js for generator scripts instead of bash

### ISSUE THREAD

**Issue by piranna** [NodeOS/NodeOS#56]
<a href="https://github.com/piranna"><img src="https://avatars.githubusercontent.com/u/532414?v=2" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [piranna](https://github.com/piranna)**
_Tuesday Aug 19, 2014 at 19:39 GMT_
_Originally opened as https://github.com/NodeOS/NodeOS-Docker/issues/10_

---

bash is so of the last century and outdated :-P Also, Javascript is easier to manage and if we are doing a Node.js based OS, is logical that their scripts are build in Javascript, and also this will help to make it self-hosted :-)

**Comment by lite20** [comment]
Status? I believe that we still use bash

**Comment by piranna** [comment]
Yes, we still use bash, and we'll still do it in the foresee future. I'm thinking to split the generic components from the Node.js parts, making that generic components independent projects that are downloaded as source deps by their actuale npm package. For example, I'm thinking to split nodeos-cross-compiler and nodeos-barebones making their more generic part fully managed by bash so they can be used by other external projects without needing to add Node.js as a dependency, and only use Node.js on their npm counterpart.

**Comment by luii** [comment]
I've created a working port of nodeos-libcurl using gulp as generator script

**Comment by piranna** [comment]
> I've created a working port of nodeos-libcurl using gulp as generator script

Not a bad thing, do you have it somewhere to comment it? since we are in Easter holidays, I'm finishing to split build steps to use BarebonesOS, that's just a generalization of NodeOS.

**Comment by luii** [comment]
Yeah here you go: https://github.com/luii/nodeos-libcurl/blob/silent-flag/gulpfile.js

Although it dont works in the install process of nodeos due to some environment things, it works in its own install process.

What i basiclly trying to say is that in the install process it somewhere overwrites/deletes my env variables and i crash the whole building of nodegit - thats the only reason i had no update on our milestone!
